RZCellSizeManager/RZCellSizeManager.m

```diff
@@ -260,33 +260,38 @@ - (void)setOverideWidth:(CGFloat)overideWidth
 
 - (void)setTraitCollection:(UITraitCollection *)traitCollection
 {
-    if ( _traitCollection != traitCollection ) {
+    if ( ![_traitCollection isEqual:traitCollection] ) {
         _traitCollection = traitCollection;
 
-        // If the trait collection is nil, or if its size classes, interface idiom, and scale are unspecified,
-        // tear down the window so we don’t waste time adding cells to a window
-        // whose trait collection won’t affect the cell’s layout.
-        BOOL traitCollectionUnspecified = (_traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassUnspecified
-                                           && _traitCollection.verticalSizeClass == UIUserInterfaceSizeClassUnspecified
-                                           && _traitCollection.userInterfaceIdiom == UIUserInterfaceIdiomUnspecified
-                                           && _traitCollection.displayScale == 0.0);
-        
-        if ( traitCollectionUnspecified ) {
-            [self.settableTraitCollectionWindow.subviews makeObjectsPerformSelector:@selector(removeFromSuperview)];
-            self.settableTraitCollectionWindow = nil;
-        }
-        else {
-            // If the window is nil, this is a no-op, which is fine, because if
-            // and when the window is created, the trait collection will get set.
-            self.settableTraitCollectionWindow.settableTraitCollection = traitCollection;
-        }
-
         // This is an over-cautious assumption. Trait collection changes do not necessarily mean that the sizes are invalidated.
         // In future, we could provide more control over cell size cache invalidation.
         [self invalidateCellSizeCache];
     }
 }
 
+#pragma mark - Getters
+
+- (SettableTraitCollectionWindow *)settableTraitCollectionWindow
+{
+    static NSMutableDictionary *settableTraitCollectionWindows = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        settableTraitCollectionWindows = [[NSMutableDictionary alloc] init];
+    });
+
+    SettableTraitCollectionWindow *settableTraitCollectionWindow = settableTraitCollectionWindows[self.traitCollection];
+    if (settableTraitCollectionWindow == nil || [settableTraitCollectionWindow isEqual:[NSNull null]]) {
```

Comment:
Ohhhh i think i must be used to it from API's and just figured that was a normal behavior